From e0b55e75e8d6ba3e85effe5f0dc37262245e47dc Mon Sep 17 00:00:00 2001
From: Tim Foster <tim.foster@joyent.com>
Date: Thu, 6 Jun 2019 13:54:25 +0000
Subject: [PATCH] OS-7823 gcc7 strap build broken by OS-7775 Reviewed by:
 Robert Mustacchi <rm@joyent.com> Reviewed by: John Levon
 <john.levon@joyent.com> Approved by: John Levon <john.levon@joyent.com>

---
 Makefile.gcc  | 8 ++++----
 gcc4/Makefile | 8 +-------
 gcc7/Makefile | 1 -
 3 files changed, 5 insertions(+), 12 deletions(-)

diff --git a/Makefile.gcc b/Makefile.gcc
index 0600969..790a806 100644
--- a/Makefile.gcc
+++ b/Makefile.gcc
@@ -21,10 +21,12 @@
 #
 SEPARATE_BUILD = yes
 
-GCC ?= /opt/local/bin/gcc
 CFLAGS = -g -O2
 PREFIX = /usr/gcc/$(GCC_VER)
 
+GCC = $(GCC.host)
+GXX = $(GXX.host)
+
 AUTOCONF_PREFIX = $(PREFIX)
 
 #
@@ -44,9 +46,7 @@ AUTOCONF_OPTS += \
 
 AUTOCONF_ENV += \
 	DESTDIR=$(DESTDIR) \
-	MAKE=$(MAKE) \
-	GCC=$(GCC.host) \
-	GXX=$(GXX.host)
+	MAKE=$(MAKE)
 
 OVERRIDES += \
 	$(AUTOCONF_CFLAGS) \
diff --git a/gcc4/Makefile b/gcc4/Makefile
index 2c67969..2ab5d0e 100644
--- a/gcc4/Makefile
+++ b/gcc4/Makefile
@@ -32,13 +32,7 @@ MPFR_TARBALL = $(MPFR_VER).tar.gz
 GMP_VER = gmp-5.1.1
 GXX = /bin/false
 CFLAGS = -nostdinc -isystem /usr/include -g -O2
-# Override these so that Makefile.gcc doesn't set them and
-# so that ../../Makefile.defs doesn't set CC.
-GCC =
 GCC.host = $(GCC4.host)
-GXX.host = $(GXX4.host)
+GXX.host = /bin/false
 
 include ../Makefile.gcc
-
-# We specifically need GCC 4.x in order to build gcc4
-AUTOCONF_ENV += CC=$(GCC4.host) CXX=$(GXX4.host)
diff --git a/gcc7/Makefile b/gcc7/Makefile
index 16a99de..0c634aa 100644
--- a/gcc7/Makefile
+++ b/gcc7/Makefile
@@ -27,7 +27,6 @@ VER = gcc-7.3.0-il-1
 MPFR_VER = mpfr-3.1.6
 GMP_VER = gmp-6.1.2
 MPC_VER = mpc-1.0.3
-GXX = /opt/local/bin/g++
 CFLAGS = -g -O2
 
 include ../Makefile.gcc
-- 
2.21.0

